core/src/main/java/org/apache/accumulo/core/rpc/clients/TServerClient.java

```diff
@@ -79,18 +83,30 @@ default Pair<String,C> getThriftServerConnection(Logger LOG, ThriftClientTypes<C
     final ZooCache zc = context.getZooCache();
 
     final List<String> serverPaths = new ArrayList<>();
-    zc.getChildren(tserverZooPath).forEach(tserverAddress -> {
-      serverPaths.add(tserverZooPath + "/" + tserverAddress);
-    });
-    if (type == ThriftClientTypes.CLIENT) {
-      zc.getChildren(sserverZooPath).forEach(sserverAddress -> {
-        serverPaths.add(sserverZooPath + "/" + sserverAddress);
-      });
+    if (type == ThriftClientTypes.CLIENT && preferredClientHost != null) {
+      // add all three paths to the set even though they may not be correct.
+      // The entire set will be checked in the code below to validate
+      // that the path is correct and the lock is held and will return the
+      // correct one.
+      serverPaths.add(tserverZooPath + "/" + preferredClientHost);
```

Comment:
Is it expected that `preferredClientHost` would include a port?  Do the paths in ZK include a port? I am not sure how the port factors into the later code that looks something up in zookeeper.